cnf-certification-test/lifecycle/suite.go

```diff
@@ -142,3 +152,31 @@ func testPodsOwnerReference(env *provider.TestEnvironment) {
 		gomega.Expect(0).To(gomega.Equal(len(badPods)))
 	})
 }
+
+func testScaling(env *provider.TestEnvironment, timeout time.Duration) {
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestDeploymentScalingIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		ginkgo.By("Testing deployment scaling")
+		defer env.SetNeedsRefresh()
+
+		if len(env.Deployments) == 0 {
+			ginkgo.Skip("No test deployments found.")
+		}
+		failedDeployments := []string{}
+		skippedDeployments := []string{}
+		for i := range env.Deployments {
+			if !scaling.TestDeploymentScaling(env.Deployments[i], env.HorizentalScaller, timeout) {
```

Comment:
for clarity, suggesting expanding the TestDeploymentScaling function here (e.g. delete the function). and add a comment explaining that if a matching hpa is present, scaling is performed using the hpa, and if not, it is performed using deployments directly

Reply:
done